gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/ClusterStatsJdbcMonitor.java

```diff
@@ -73,8 +81,16 @@ public ClusterStats monitor(ProxyBackendConfiguration backend)
         }
 
         try (Connection conn = DriverManager.getConnection(jdbcUrl, properties)) {
-            PreparedStatement stmt = SimpleTimeLimiter.create(Executors.newSingleThreadExecutor())
-                    .callWithTimeout(() -> conn.prepareStatement(STATE_QUERY), 10, TimeUnit.SECONDS);
+            networkTimeoutMillis.ifPresent(t -> {
+                try {
+                    conn.setNetworkTimeout(Executors.newSingleThreadExecutor(), 10); //executor is unused in TrinoConnection.setNetworkTimeout
+                }
+                catch (SQLException e) {
+                    log.error(e, "Could not set network timeout for JDBC health monitor");
+                }
+            });
+            PreparedStatement stmt = SimpleTimeLimiter.create(Executors.newSingleThreadExecutor()).callWithTimeout(
+                    () -> conn.prepareStatement(STATE_QUERY), max(10, 1000 * networkTimeoutMillis.orElse(0)) + 1, TimeUnit.SECONDS);
```

Comment:
we should add a comment here to explain the 10 seconds magic number here. Basically this is querying the `system.runtime` table and we don't expect more than 10 seconds, and hence it is "hardcoded" here and not configurable. There were some asks to make it configurable, which is something we can do, but it doesn't add much value since 10 seconds should be more than enough

Reply:
This is making it configurable - I was using 10 as the default, but did so in a confusing way, and screwed up the conversion from millis to seconds to boot.

Reply:
this was reverted since the network config was superseded by https://github.com/trinodb/trino-gateway/pull/436